dlrs/main/classes/RollupSummariesSelector.cls

```diff
@@ -338,23 +372,52 @@ public class RollupSummariesSelector {
       List<String> calculationModeNames = new List<String>();
       for (RollupSummaries.CalculationMode calculationMode : calculationModes)
         calculationModeNames.add(calculationMode.name());
-      return RollupSummary.toList(
-        Database.query(
-          newQueryFactory()
-            .setCondition(
-              'CalculationMode__c in :calculationModeNames and ChildObject__c in :childObjectNames and Active__c = true'
-            )
-            .addOrdering(
-              LookupRollupSummary2__mdt.ParentObject__c,
-              fflib_QueryFactory.SortOrder.ASCENDING
-            )
-            .addOrdering(
-              LookupRollupSummary2__mdt.RelationshipField__c,
-              fflib_QueryFactory.SortOrder.ASCENDING
-            )
-            .toSOQL()
-        )
-      );
+      if (this.forEdit) {
+        return RollupSummary.toList(
+          this.getLookupRollupSummaries(
+            newQueryFactory()
+              .setCondition(
+                'CalculationMode__c in :tmpVar1 and ChildObject__c in :tmpVar2 and Active__c = true'
+              )
+              .addOrdering(
+                LookupRollupSummary2__mdt.ParentObject__c,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .addOrdering(
+                LookupRollupSummary2__mdt.RelationshipField__c,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .toSOQL(),
+            new List<Object>{ calculationModeNames, childObjectNames }
+          )
+        );
+      } else {
+        List<RollupSummary> summaries = RollupSummary.toList(
+          this.getLookupRollupSummaries()
+        );
+
+        List<RollupSummary> filteredSummaries = new List<RollupSummary>();
+
+        for (RollupSummary summary : summaries) {
+          SObject record = summary.Record;
+          if (
+            record != null &&
+            (Boolean) record.get('Active__c') &&
+            calculationModeNames.contains(
```

Comment:
@afawcett we talked about this and thought it was meant to be `.equals` and not `.contains`, but reading a little more it appears that .contains is correct because `calculationModeNames` and `childObjectNames` are Lists, not Strings.

Adding this comment as an FYI to Andy - nothing to do here for the code review.